firmware/shared/src/hw/hw_spi.c

```diff
@@ -22,15 +23,28 @@ bool hw_spi_transmitAndReceive(
 {
     bool status = false;
 
+    const uint16_t combined_size = tx_buffer_size + rx_buffer_size;
+
+    // HAL_SPI_TransmitRecieve requires tx buffer and rx buffer to be of size equal to number of bytes to transmit and
+    // recieve
+    uint8_t padded_tx_buffer[combined_size];
+    uint8_t padded_rx_buffer[combined_size];
+
+    // Copy tx_buffer into beginning of larger padded_tx_buffer
+    memcpy(padded_tx_buffer, tx_buffer, tx_buffer_size);
+
     hw_spi_setNssLow(spi);
 
-    if (HAL_SPI_Transmit(spi->spi_handle, tx_buffer, tx_buffer_size, spi->timeout_ms) == HAL_OK)
-    {
-        status = HAL_SPI_Receive(spi->spi_handle, rx_buffer, rx_buffer_size, spi->timeout_ms) == HAL_OK;
-    }
+    status = HAL_SPI_TransmitReceive(
```

Comment:
HAL_SPI_Transmit and HAL_SPI_Recieve both internally call HAL_SPI_TransmitReceive, cutting out the middleman and calling it directly is more efficient, even if it requires a bit of our own glue logic.